docs/template.md

```diff
@@ -3,4 +3,80 @@ title: "Template"
+
+Template is an independent NodeJS project that’s not related to the Generator repository. AsyncAPI templates are managed, released and published separately. Generator uses the Arborist library. Arborist connects both Template and Generator, It is also the dependency tree manager for npm. 
+
+Arborist helps the generator fetch source code or hooks of the template and use that for the generation process. It also means templates are a set of customly designed files which can be stored anywhere like–in npm, on local during the development process, or just as a github repository. You can do anything that is already possible with `npm install`.
+
+The following section talks about the minimum components your template should have in any given situation:
+
+# Minimum for your template
+
+The most basic template must have the following:
+
+### 1. `template` directory
+
+This is where you keep the files that determine the output that will be generated. In other words, the Generator processes all the files stored in this directory.
+
+```js
+import { File, Text } from "@asyncapi/generator-react-sdk";
+
+export default function({ asyncapi, params, originalAsyncAPI }) {
+return (
+    <File name="asyncapi.md">
+    <Text>This is a markdown file for my application.</Text>
+    <Text>App name is: **{ asyncapi.info().title() }**</Text>
+    </File>
+);
+}
+```
+
+> Code sample developed is stored under template/index.js folder
+
+### 2. `package.json` file
+
+This file is necessary even if your template doesn't need any external dependencies. Before the generation process runs, the Generator must install the template into its dependencies, and `package.json` is necessary to identify the template name.
+
+Following block shows an example `package.json` file and some select dependencies:
+
+```json
+"generator": {
+    "renderer": "react",
+    "parameters": {
+      "server": {
+        "description": "The server you want to use in the code.",
+        "required": true
+      },
+
+"dependencies": {
+    "@asyncapi/generator-filters": "^2.1.0",
+    "@asyncapi/generator-hooks": "^0.1.0",
+    "@asyncapi/generator-react-sdk": "^0.2.16"
+  },
+```
+
+> Whenever you make a change to package.json make sure you perform an update with `npm install` to synchronize with package-lock.json and validate if the file is not broken after your changes.
+
+Templates are reusable. Same template can be modified and used for different purposes. [Hooks](hooks.md) allow Templates to perform multiple tasks. Hooks are fractions of code that can be added to your Template. They can be plugged-in onto the generation process to perform multiple actions. For example, we can configure a hook that generates a pdf after the generation process is completed. 
```

Comment:
```suggestion
Templates are reusable. Same template can be modified and used for different purposes. [Hooks](hooks.md) enable templates to perform multiple tasks. Hooks are fractions of code that can be added to your Template. They can be plugged-in onto the generation process to perform multiple actions. For example, we can configure a hook that generates a pdf after the generation process is completed. 
```

> Templates are reusable. Same template can be modified and used for different purposes.

what do you mean?

btw, hooks should be a separate section or document 🤔 
I'm just thinking if we should not have separate document:
- about template for end user, so they understand what they are, how to use them, that there are properties
- about template for template developer, because concept of `hook` is useful for them only. End user do not have to know what hook is as they work behind the scenes by default, or through parameters, but still users has no idea, and do not need to know what that is